src/rp2_common/pico_multicore/include/pico/multicore.h

```diff
@@ -130,9 +130,12 @@ static inline void multicore_fifo_drain(void) {
 
 /*! \brief Clear FIFO interrupt
  *  \ingroup multicore_fifo
+ *
+ * Note that this only clears an interrupt that was caused by the ROE or WOF flags.
+ * To clear the VLD flag you need to use one of the 'pop' or 'drain' functions.
 */
```

Comment:
yeah; i guess we should rename this function as it doesn't really do what it says; perhaps `multicore_clear_sticky_errors`

Note we need a little documentation update here (this file) anyway to discourage use of the FIFOs

Reply:
> i guess we should rename this function

That would be a backwards-incompatible change, which according to semver conventions would necessitate a version-bump to 2.0.0

> Note we need a little documentation update here (this file) anyway to discourage use of the FIFOs

Sounds like a Graham-job to me! :wink:  I know very little about the FIFOs (and even less about multicore in general).